src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -281,18 +248,58 @@ private void AnalyzeCompilationStart(CompilationStartAnalysisContext context)
                     }
                     else
                     {
+                        // Prevent use of unassigned variables error
                         return;
                     }
 
-                    context.ReportDiagnostic(
-                        invocation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
+                    context.ReportDiagnostic(awaitOperation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
```

Comment:
Suggest making the change below and reverting all test changes in the last commit:

```suggestion
                    context.ReportDiagnostic(awaitOperation.Operation.CreateDiagnostic(rule, ruleMessageMethod, ruleMessagePreferredMethod));
```